eng/scripts/live-test-resource-cleanup.ps1

```diff
@@ -38,58 +48,167 @@ param (
+      $ResourceGroup | Update-AzTag -Operation Merge -Tag @{ DeleteAfter = $deleteAfter }
     }
-    Write-Verbose "Deleting group: $($rg.ResourceGroupName)"
-    Write-Verbose "  tags $($rg.Tags | ConvertTo-Json -Compress)"
-    Write-Host ($rg | Remove-AzResourceGroup -Force -AsJob).Name
   }
 }
 
-# Purge all the purgeable resources and get a list of resources (as a collection) we need to follow-up on.
-Write-Host "Attempting to purge $($purgeableResources.Count) resources."
-$failedResources = @(Remove-PurgeableResources $purgeableResources -PassThru)
-if ($failedResources) {
-  Write-Warning "Timed out deleting the following $($failedResources.Count) resources. Please file an IcM ticket per resource type."
-  $failedResources | Sort-Object AzsdkResourceType, AzsdkName | Format-Table -Property @{l='Type'; e={$_.AzsdkResourceType}}, @{l='Name'; e={$_.AzsdkName}}
+function HasDeleteLock() {
+  return $false
+}
+
+function DeleteOrUpdateResourceGroups() {
+  [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+  param()
+
+  Write-Verbose "Fetching groups"
+  $allGroups = @(Get-AzResourceGroup)
+  $toDelete = @()
+  $toUpdate = @()
+  Write-Host "Total Resource Groups: $($allGroups.Count)"
+
+  foreach ($rg in $allGroups) {
+    if (HasExpiredDeleteAfterTag $rg) {
+      $toDelete += $rg
+    } elseif (HasValidOwnerTag $rg -or HasDeleteLock $rg) {
+      continue
+    } else {
+      $toUpdate += $rg
+    }
+  }
+
+  foreach ($rg in $toUpdate) {
+    FindOrCreateDeleteAfterTag $rg
+  }
+
+  # Get purgeable resources already in a deleted state.
+  $purgeableResources = @(Get-PurgeableResources)
+
+  Write-Host "Total Resource Groups To Delete: $($toDelete.Count)"
+  foreach ($rg in $toDelete)
+  {
+    if ($Force -or $PSCmdlet.ShouldProcess("$($rg.ResourceGroupName) (UTC: $($rg.Tags.DeleteAfter))", "Delete Group")) {
+      # Add purgeable resources that will be deleted with the resource group to the collection.
+      $purgeableResourcesFromRG = Get-PurgeableGroupResources $rg.ResourceGroupName
+
+      if ($purgeableResourcesFromRG) {
+        $purgeableResources += $purgeableResourcesFromRG
+        Write-Verbose "Found $($purgeableResourcesFromRG.Count) potentially purgeable resources in resource group $($rg.ResourceGroupName)"
+      }
+      Write-Verbose "Deleting group: $($rg.ResourceGroupName)"
+      Write-Verbose "  tags $($rg.Tags | ConvertTo-Json -Compress)"
+      # Write-Host ($rg | Remove-AzResourceGroup -Force -AsJob).Name
+    }
+  }
+
+  if ($Force -or $PSCmdlet.ShouldProcess("$($purgeableResources.VaultName)", "Delete Purgeable Resources")) {
+    # Purge all the purgeable resources and get a list of resources (as a collection) we need to follow-up on.
+    Write-Host "Attempting to purge $($purgeableResources.Count) resources."
+    # $failedResources = @(Remove-PurgeableResources $purgeableResources -PassThru)
```

Comment:
Not sure if purging the resources is intentionally disabled... it was my assumption in the question farther up about fetching purgable resources if not deleting them later.